pageserver/src/tenant.rs

```diff
@@ -5617,29 +5617,278 @@ mod tests {
+        let harness = TenantHarness::create("aux_file_policy_is_not_switched_back")?;
+        assert_eq!(
+            harness.tenant_conf.switch_aux_file_policy,
+            AuxFilePolicy::V1
+        );
+        let (tenant, ctx) = harness.load().await;
+
+        let mut lsn = Lsn(0x08);
+
+        let tline: Arc<Timeline> = tenant
+            .create_test_timeline(TIMELINE_ID, lsn, DEFAULT_PG_VERSION, &ctx)
             .await?;
-        assert_eq!(tline2.last_aux_file_policy.load(), Some(AuxFilePolicy::V2));
+
+        assert_eq!(
+            tline.last_aux_file_policy.load(),
+            None,
+            "this should be V1 but there's no Default impl"
+        );
+
+        {
+            lsn += 8;
+            let mut modification = tline.begin_modification(lsn);
+            modification
+                .put_file("pg_logical/mappings/test1", b"first", &ctx)
+                .await?;
+            modification.commit(&ctx).await?;
+        }
+
+        // there is no tenant manager to pass the configuration through, so lets mimic it
+        tenant.set_new_location_config(
+            AttachedTenantConf::try_from(LocationConf::attached_single(
+                TenantConfOpt {
+                    switch_aux_file_policy: Some(AuxFilePolicy::CrossValidation),
+                    ..Default::default()
+                },
+                tenant.generation,
+                &pageserver_api::models::ShardParameters::default(),
+            ))
+            .unwrap(),
+        );
+
+        assert_eq!(
+            tline.get_switch_aux_file_policy(),
+            AuxFilePolicy::CrossValidation,
+            "wanted state has been updated"
+        );
+        assert_eq!(
+            tline.last_aux_file_policy.load(),
+            Some(AuxFilePolicy::V1),
+            "dirty index_part.json reflected state is yet to be updated"
+        );
+
+        // we can still read the auxfile v1 before we ingest anything new
+        let files = tline.list_aux_files(lsn, &ctx).await.unwrap();
+        assert_eq!(
+            files.get("pg_logical/mappings/test1"),
+            Some(&bytes::Bytes::from_static(b"first"))
+        );
+
+        {
+            lsn += 8;
+            let mut modification = tline.begin_modification(lsn);
+            modification
+                .put_file("pg_logical/mappings/test2", b"second", &ctx)
+                .await?;
+            modification.commit(&ctx).await?;
+        }
+
+        assert_eq!(
+            tline.last_aux_file_policy.load(),
+            Some(AuxFilePolicy::CrossValidation),
+            "ingesting a file should apply the wanted switch state when applicable"
+        );
+
+        let files = tline.list_aux_files(lsn, &ctx).await.unwrap();
+        assert_eq!(
+            files.get("pg_logical/mappings/test1"),
+            Some(&bytes::Bytes::from_static(b"first")),
+            "V1 file is readable in cross validation even though it is not found as V2"
+        );
```

Comment:
This now fails because the test1 is not found as v2. I think that should be accepted, but perhaps it shouldn't be? Without a test, we cannot know.

Reply:
Fixed with [61676f3](https://github.com/neondatabase/neon/pull/7781/commits/61676f332fde5f93a38a5651bf53cd8049f84238), then the next problem is crossvalidation: [156348e](https://github.com/neondatabase/neon/pull/7781/commits/156348e9790f165e9e03b3628bd1ba56f6517fc3)